web-app/src/App.jsx

```diff
@@ -1,24 +1,46 @@
 import React from 'react';
-import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { Route, Switch, withRouter, Redirect } from 'react-router-dom';
 import { Dashboard, Login, Admin, User } from './pages';
 import { ThemeProvider } from 'styled-components';
+import { PropTypes as PT } from 'prop-types';
 import { theme } from './styled';
+import Layout from './hoc/Layout';
 
 class App extends React.Component {
+  static propTypes = {
+    drawerOpen: PT.bool,
+    history: PT.object
+  };
+
   render() {
+    let isAuthenticated = localStorage.getItem('id_token') ? true : false;
+    let routes = (
```

Comment:
I'd use an if...else here for assigning the routes content, just to avoid running unrequired code.